src/systemd/meson.build

```diff
@@ -62,7 +65,7 @@ if cxx.found()
         endif
 endif
 
-foreach header : _systemd_headers
+foreach header : _systemd_headers + _not_installed_headers
```

Comment:
@yuwata thank you! This certainly does the trick. I've just reverted b62f9008668a5330c61b4 to see whether I would be able to spot the regression introduced in d89a400ed664e0b57fd8667 and I got a lot of failed tests saying that 
```
In file included from <command-line>:32:0:
/home/vagrant/systemd/src/systemd/sd-dhcp6-client.h:28:10: fatal error: sparse-endian.h: No such file or directory
 #include "sparse-endian.h"
          ^~~~~~~~~~~~~~~~~
compilation terminated.
```
Pretty cool! It makes it much harder for me to screw something up again :-) One question I have, though, is why that block was commented out. I'll try to take a look at the history.

Reply:
> why that block was commented out

The commented out headers are not installed. So, we did not test them.

Reply:
That's right. But they're still in a kind of intermediate state in the sense that although they aren't installed, some other projects apparently use them. It probably deserves a comment. @poettering @thom311  could you comment on the plans to expose the API that were mentioned in #10277 so that in the future it would be a little bit easier to understand why the change was made?

Reply:
On the other hand, in theory, "Follow-up for #10277" in the commit message should be enough for a careful reader to figure out what's going on :-) LGTM! I'd wait for someone else to review it, though.

Reply:
@evverx i figure we should make sd-utf8 and sd-path public sooner or later, they should be OK the way they are.

And the network stuff I am not so sure about yet. if we make them public they should be separate from libsystemd